File: backend/src/Entity/MailScan.php
<?php

namespace App\Entity;

use App\Repository\MailScanRepository;
use Doctrine\DBAL\Types\Types;
use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Uid\Uuid;

#[ORM\Entity(repositoryClass: MailScanRepository::class)]
class MailScan
{
    #[ORM\Id]
    #[ORM\Column(type: 'uuid', unique: true)]
    #[ORM\GeneratedValue(strategy: 'CUSTOM')]
    #[ORM\CustomIdGenerator(class: 'doctrine.uuid_generator')]
    private ?Uuid $id = null;

    #[ORM\ManyToOne(targetEntity: User::class)]
    #[ORM\JoinColumn(nullable: false)]
    private ?User $user = null;

    #[ORM\Column]
    private ?\DateTimeImmutable $scannedAt = null;

    #[ORM\Column(nullable: true)]
    private ?int $mailsAnalyzed = null;

    #[ORM\Column(nullable: true)]
    private ?int $matchesFound = null;

    #[ORM\Column(length: 20)]
    private string $status = 'success'; // 'success' | 'error'

    #[ORM\Column(type: Types::TEXT, nullable: true)]
    private ?string $errorMessage = null;

    public function __construct()
    {
        $this->scannedAt = new \DateTimeImmutable();
    }

    public function getId(): ?Uuid { return $this->id; }

    public function getUser(): ?User { return $this->user; }
    public function setUser(?User $v): static { $this->user = $v; return $this; }

    public function getScannedAt(): ?\DateTimeImmutable { return $this->scannedAt; }

    public function getMailsAnalyzed(): ?int { return $this->mailsAnalyzed; }
    public function setMailsAnalyzed(?int $v): static { $this->mailsAnalyzed = $v; return $this; }

    public function getMatchesFound(): ?int { return $this->matchesFound; }
    public function setMatchesFound(?int $v): static { $this->matchesFound = $v; return $this; }

    public function getStatus(): string { return $this->status; }
    public function setStatus(string $v): static { $this->status = $v; return $this; }

    public function getErrorMessage(): ?string { return $this->errorMessage; }
    public function setErrorMessage(?string $v): static { $this->errorMessage = $v; return $this; }
}

File: backend-java/src/main/java/com/jobdesk/domain/MailScan.java
package com.jobdesk.domain;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.PrePersist;
import jakarta.persistence.Table;

import java.time.LocalDateTime;
import java.util.UUID;

@Entity
@Table(name = "mail_scan")
public class MailScan {

    @Id
    private UUID id;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    @Column(nullable = false)
    private LocalDateTime scannedAt;

    private Integer mailsAnalyzed;

    private Integer matchesFound;

    @Column(length = 20, nullable = false)
    private String status = "success";

    @Column(columnDefinition = "text")
    private String errorMessage;

    @PrePersist
    void onPrePersist() {
        if (id == null) {
            id = UUID.randomUUID();
        }
        if (scannedAt == null) {
            scannedAt = LocalDateTime.now();
        }
    }

    public UUID getId() {
        return id;
    }

    public User getUser() {
        return user;
    }

    public void setUser(User user) {
        this.user = user;
    }

    public LocalDateTime getScannedAt() {
        return scannedAt;
    }

    public void setScannedAt(LocalDateTime scannedAt) {
        this.scannedAt = scannedAt;
    }

    public Integer getMailsAnalyzed() {
        return mailsAnalyzed;
    }

    public void setMailsAnalyzed(Integer mailsAnalyzed) {
        this.mailsAnalyzed = mailsAnalyzed;
    }

    public Integer getMatchesFound() {
        return matchesFound;
    }

    public void setMatchesFound(Integer matchesFound) {
        this.matchesFound = matchesFound;
    }

    public String getStatus() {
        return status;
    }

    public void setStatus(String status) {
        this.status = status;
    }

    public String getErrorMessage() {
        return errorMessage;
    }

    public void setErrorMessage(String errorMessage) {
        this.errorMessage = errorMessage;
    }
}
